homeassistant/helpers/selector.py

```diff
@@ -295,25 +300,179 @@ class StringSelector(Selector):
 
     selector_type = "text"
 
-    CONFIG_SCHEMA = vol.Schema({vol.Optional("multiline", default=False): bool})
+    STRING_TYPES = [
+        "number",
+        "text",
+        "search",
+        "tel",
+        "url",
+        "email",
+        "password",
+        "date",
+        "month",
+        "week",
+        "time",
+        "datetime-local",
+        "color",
+    ]
+    CONFIG_SCHEMA = vol.Schema(
+        {
+            vol.Optional("multiline", default=False): bool,
+            vol.Optional("suffix"): str,
+            vol.Optional("type"): vol.In(STRING_TYPES),
+        }
+    )
 
     def __call__(self, data: Any) -> str:
         """Validate the passed selection."""
         text = cv.string(data)
         return text
 
 
+select_option = vol.All(
+    dict,
+    vol.Schema(
+        {
+            vol.Required("value"): str,
+            vol.Required("label"): str,
+        }
+    ),
+)
+
+
 @SELECTORS.register("select")
 class SelectSelector(Selector):
     """Selector for an single-choice input select."""
 
     selector_type = "select"
 
     CONFIG_SCHEMA = vol.Schema(
-        {vol.Required("options"): vol.All([str], vol.Length(min=1))}
+        {
+            vol.Required("options"): vol.All(
+                vol.Any([str], [select_option]), vol.Length(1)
+            )
+        }
     )
 
     def __call__(self, data: Any) -> Any:
         """Validate the passed selection."""
-        selected_option = vol.In(self.config["options"])(cv.string(data))
-        return selected_option
+        if isinstance(self.config["options"][0], str):
+            options = self.config["options"]
+        else:
+            options = [option["value"] for option in self.config["options"]]
+        return vol.In(options)(cv.string(data))
+
+
+@SELECTORS.register("attribute")
+class AttributeSelector(Selector):
+    """Selector for an entity attribute."""
+
+    selector_type = "attribute"
+
+    CONFIG_SCHEMA = vol.Schema({vol.Required("entity_id"): cv.entity_id})
+
+    def __call__(self, data: Any) -> str:
+        """Validate the passed selection."""
+        return cv.string(data)
```

Comment:
This does some stringifying, while rejecting `None` etc., maybe we should only allow actual `str` instances?